x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlMemoryAutoscalingCapacity.java

```diff
@@ -25,6 +25,10 @@ public String toString() {
         return "MlMemoryAutoscalingCapacity{" + "nodeSize=" + nodeSize + ", tierSize=" + tierSize + ", reason='" + reason + '\'' + '}';
     }
 
+    public boolean isUndetermined() {
+        return nodeSize == null && tierSize == null;
```

Comment:
Hopefully it's not possible that we ever have a tier size but no node size or vice versa.

If you add the following assertion here and in the constructor of `Builder` do all the tests still pass?

```
    assert (nodeSize == null) == (tierSize == null) : "nodeSize " + nodeSize + " tierSize " + tierSize;
```

Reply:
Yep, good one.